Clean up and Document Placement

saaramahmoudi opened the pull request:
#### Description
Small PR; placer seed was always printed as 0 in vpr.out, since It was printed before It was set. Hence, the default value regardless of the command-line was printed. Set it before printing. 


#### Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed

saaramahmoudi commented:
@vaughnbetz

saaramahmoudi commented:
@vaughnbetz can you please publish your review so I can take actions?

vaughnbetz commented on vpr/src/base/SetupVPR.cpp:
```diff
@@ -666,6 +666,8 @@ static void SetupPlacerOpts(const t_options& Options, t_placer_opts* PlacerOpts)
     PlacerOpts->floorplan_num_horizontal_partitions = Options.floorplan_num_horizontal_partitions;
     PlacerOpts->floorplan_num_vertical_partitions = Options.floorplan_num_vertical_partitions;
+
+    PlacerOpts->seed = Options.Seed;
```
@saaramahmoudi : placerOpts->seed setting line is now removed from the placer. If the two documentation updates can be made we can merge this. Or if that will take a while please create an issue to track them and I'll merge this.

saaramahmoudi commented:
@vaughnbetz The RLPlace command-line options are now added to the documentation. I also added NoC options to documentation since they weren't available either and fixed some traffic flow typos.

vaughnbetz commented on doc/src/vpr/command_line_usage.rst:
```diff
@@ -920,6 +980,82 @@ The following options are only valid when the placement engine is in timing-driv
 
     Name of the post-placement timing report file to generate (not generated if unspecfied).
 
+
```
I know @Srivat97 has a PR that's about to land that documents the NoC options too. @Srivat97 : I suggest landing yours and then Sara can enhance any NoC documentation that doesn't capture everything she has here.

vaughnbetz commented on doc/src/vpr/command_line_usage.rst:
```diff
@@ -821,6 +821,66 @@ Setting any of the following options selects `Dusty's annealing schedule <dusty_
+.. option:: --place_cost_exp <float>
+
+    Wiring cost is divided by the average channel width over a net's bounding box
+    taken to this exponent.Only impacts devices with different channel widths in 
```
Add a space after "."

vaughnbetz commented on doc/src/vpr/command_line_usage.rst:
```diff
@@ -821,6 +821,66 @@ Setting any of the following options selects `Dusty's annealing schedule <dusty_
+.. option:: --RL_agent_placement {on | off}
+
+    Uses a Reinforcement Learning (RL) agent in choosing the appropiate move type in placement.
+    It activates the RL agent placement instead of using fixed probability for each move type.
```
"a" fixed probability for each

vaughnbetz commented on doc/src/vpr/command_line_usage.rst:
```diff
@@ -821,6 +821,66 @@ Setting any of the following options selects `Dusty's annealing schedule <dusty_
+
+.. option:: --place_agent_multistate {on | off}
+
+    Enable multistate agent in the placement. A second state will be activated late in
```
"Enable a multistate agent during RL placement."  (first sentence).

vaughnbetz commented on doc/src/vpr/command_line_usage.rst:
```diff
@@ -821,6 +821,66 @@ Setting any of the following options selects `Dusty's annealing schedule <dusty_
+
+.. option:: --place_agent_epsilon <float>
+
+    Placement RL agent's epsilon for epsilon-greedy agent. Epsilon represents
```
"for the epsilon-greedy agent."

vaughnbetz commented on doc/src/vpr/command_line_usage.rst:
"by the placement RL agent to learn the best action"  (missing a couple of "the" )

vaughnbetz commented on doc/src/vpr/command_line_usage.rst:
"The RL agent exploration space can either be based only on move types or can also consider the block type being moved."

vaughnbetz reviewed (changes requested):
Looks good, except for a few wording nits.  
Also, @Srivat97 has simultaneously been making a PR for NoC documentation so you'll need to sync up with him on that part.

saaramahmoudi commented:
> Looks good, except for a few wording nits. Also, @Srivat97 has simultaneously been making a PR for NoC documentation so you'll need to sync up with him on that part.

I don't see a PR from @Srivat97. Can you put the PR's name here and I can merge the branches and land both together?

vaughnbetz commented:
@Srivat97 : probably you haven't made a PR yet (@saaramahmoudi : Srivatsan showed me the documentation in our meeting today, but I don't see the PR either). Can you make a PR @Srivat97 ?

saaramahmoudi commented:
@vaughnbetz I just did a small clean-up on NoC router description in vpr.out, I moved printing of router connections in the architecture to echo file instead of printing 100 lines in the vpr.out. Could you please take a look at it to see if that sounds good to you?

vaughnbetz commented:
Looks good, thanks.  Will merge after talking to Srivatsan today.

saaramahmoudi commented:
@vaughnbetz ready to be merged.

Srivat97 commented:
@vaughnbetz , @saaramahmoudi : Sorry for the delayed response. You can go ahead and merge this PR first. I'll handle the conflicts on my end.
